spring-cloud-gcp-samples/spring-cloud-gcp-vision-ocr-demo/README.adoc

```diff
@@ -0,0 +1,107 @@
+= Spring GCP Cloud Vision OCR Demo Application
+
+This demonstrates how to run Google Cloud Vision https://cloud.google.com/vision/docs/ocr[Optical Character Recognition (OCR)] analysis on PDF and TIFF documents.
+OCR is a process which allows you to detect and extract the text from digital documents.
+
+This demo application allows you to upload documents to a specified https://cloud.google.com/storage/docs/key-terms#buckets[Google Cloud Storage bucket] and run OCR on those documents you uploaded.
+After your documents are uploaded and processed, the application will allow you to access the extracted text from each document.
+
+The main work of running OCR in the application occurs in the `submitDocument` method in `WebController.java`:
+
+[source,java]
+----
+@PostMapping("/submitDocument")
+public ModelAndView submitDocument(@RequestParam("documentUrl") String documentUrl) throws IOException {
+
+  // Uploads the document to the GCS bucket
+  Resource documentResource = resourceLoader.getResource(documentUrl);
+  BlobId outputBlobId = BlobId.of(ocrBucket, documentResource.getFilename());
+  BlobInfo blobInfo =
+      BlobInfo.newBuilder(outputBlobId)
+          .setContentType(getFileType(documentResource))
+          .build();
+
+  try (WriteChannel writer = storage.writer(blobInfo)) {
+    ByteStreams.copy(documentResource.getInputStream(), Channels.newOutputStream(writer));
```

Comment:
Yeah that does work; however currently our `GoogleStorageResource` class doesn't yet let you control the type of blob you create; like this:

```
BlobInfo.newBuilder(outputBlobId)
          .setContentType(getFileType(documentResource))
          .build();
```

Currently all files created through GoogleStorageResource defaults to type: "application/octet-stream". Unfortunately this information is needed b/c you otherwise get this error when submitting the document to vision API: 

```
INVALID_ARGUMENT: At this time, the only MIME types supported are 'application/pdf', 'image/gif' and 'image/tiff'.
```

Reply:
Can you file a bug regarding this so we can fix it going forward? Thanks!

Reply:
Done: https://github.com/spring-cloud/spring-cloud-gcp/issues/1603